src/main/java/seedu/address/logic/filters/CarsOwnedFilter.java

```diff
@@ -0,0 +1,35 @@
+package seedu.address.logic.filters;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.customer.Customer;
+
+/**
+ * Filters CarsOwned based on user specified filters carBrand, carType.
+ */
+public class CarsOwnedFilter extends CarFilter {
+    /**
+     * Creates a filter for CarsOwned.
+     *
+     * @param filterString
+     */
+    public CarsOwnedFilter(String filterString) {
+        super(filterString.trim());
+        requireNonNull(filterString);
+    }
+
+    /**
+     * Tests each car in CarsOwned, with the specified filters.
+     *
+     * @param customer
+     * @return
+     */
+    @Override
+    public boolean test(Customer customer) {
+        requireNonNull(customer);
+        return customer.getCarsOwned()
+                .keySet()
+                .parallelStream()
+                .anyMatch(x -> carFilterPredicate(x, parseCar(filterString)));
```

Comment:
Could replace `x` with `customerCar`, but its not exactly block worthy this close to deadline. Fix soon during 1.4 please.